deepmd/dpmodel/descriptor/descriptor.py

```diff
@@ -129,3 +129,29 @@ def call(
     @abstractmethod
     def has_message_passing(self) -> bool:
         """Returns whether the descriptor block has message passing."""
+
+
+def extend_descrpt_stat(des, type_map, des_with_stat=None):
+    """
+    Extend the statistics of a descriptor block with types in type_map.
+
+    Parameters
+    ----------
+    des : DescriptorBlock
+        The descriptor block to be extended.
+    type_map : List[str]
+        The name of each type of atoms to be extended.
+    des_with_stat : DescriptorBlock, Optional
+        The descriptor block has additional statistics in type_map.
+        If None, the default statistics will be used. Otherwise, the statistics provided in this DescriptorBlock will be used.
+
+    """
+    if des_with_stat is not None:
+        extend_davg = des_with_stat["davg"]
+        extend_dstd = des_with_stat["dstd"]
+    else:
+        extend_shape = [len(type_map), *list(des["davg"].shape[1:])]
+        extend_davg = np.zeros(extend_shape, dtype=des["davg"].dtype)
+        extend_dstd = np.ones(extend_shape, dtype=des["dstd"].dtype)
+    des["davg"] = np.concatenate([des["davg"], extend_davg], axis=0)
+    des["dstd"] = np.concatenate([des["dstd"], extend_dstd], axis=0)
```

Comment:
you append the stats of the `des_with_stat` to the statistics of `self`. 
1. do you assume that the type map of `des_with_stat` is the new type map? 
2. is `type_map` argument assumed to be the new type map? 
3. please provide a better doc for the assumptions. 
4. please also not that the behavior should work with `get_index_between_two_maps`

Reply:
1.2. yes
3.4. docs updated

This behaivior is developed for extending new types unseen in old type_map. 

In this situation, some type related params in the model, such as statistics, need to be extended to have a length of `len(old_map) + len(new_map)` in the type related dimension. 

`get_index_between_two_maps` will generate negative indices `i - len(new_map)` if `new_map[i]` is not in the `old_map`.
Then positive indices from 0 to `len(old_map) - 1` will select old params of types in `old_map`, while negative indices from `-len(new_map)` to -1 will select new params of types in `new_map`.